engines/gob/surface.cpp

```diff
@@ -854,29 +909,70 @@ void Surface::blitToScreen(uint16 left, uint16 top, uint16 right, uint16 bottom,
 	g_system->copyRectToScreen(src, _width * _bpp, x, y, width, height);
 }
 
-bool Surface::loadImage(Common::SeekableReadStream &stream) {
+bool Surface::getImageInfo(Common::SeekableReadStream &stream, uint32 &width, uint32 &height, uint32 &bpp) {
 	ImageType type = identifyImage(stream);
 	if (type == kImageTypeNone)
 		return false;
 
-	return loadImage(stream, type);
+	Common::ScopedPtr<Image::ImageDecoder> decoder;
+	switch (type) {
+	case kImageTypeTGA:
+		decoder.reset(new Image::TGADecoder());
+		break;
+	case kImageTypeIFF:
+		decoder.reset(new Image::IFFDecoder());
+		break;
+	case kImageTypeBRC:
+		warning("Surface::getImageInfo(): BRC images are not supported");
```

Comment:
I believe now you added the decoder?

Reply:
Yes indeed! Filled the case kImageTypeBRC